File: 프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.js
function solution(s) {
    let maxLen = 1;
    function expand (left,right) {
        while(left>=0 && right<s.length && s[left]===s[right]){
            maxLen = Math.max(maxLen, right-left+1);
            left--;
            right++;
        }
    }

    for (let i = 0; i < s.length; i++) {
        expand(i, i);     // 홀수 길이 팰린드롬
        expand(i, i + 1); // 짝수 길이 팰린드롬
    }

    return maxLen;
}

File: 프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.java
class Solution
{
    static int answer;
    public int solution(String s)
    {
        answer = 0;
        String[] str = s.split("");
        for(int i=0; i<str.length; i++){
            //i가 기준점
            answer = Math.max(answer,pend(str,i,i));
            answer = Math.max(answer,pend(str,i,i+1));
        }

        return answer;
    }
    static int pend(String[] str,int start, int end){
        int len = 0;
        while(start>=0&&end<str.length&&str[start].equals(str[end])){
            len = Math.max(len, end-start+1);
            start--;
            end++;
        }
        return len;
    }
}